lib/Models/CswCatalogGroup.js

```
@@ -447,13 +481,29 @@ Otherwise, if reloading doesn\'t fix it, please report the problem by sending an
 
 };
 
+function createWpsItemFromUri(terria, uri, title) {
+    var wps = new WebProcessingServiceCatalogGroup(terria);
+    wps.name = title;
+    wps.url = uri;
+
+    return wps.load().then(function() {
+        return wps;
+    });
+}
+
+function addProcessesToGroup(wpsPromise, wpsGroup) {
+    wpsPromise.then(function(wpsItem) {
```

Without an `otherwise`, a WPS load error will silently fail telling the user about it.